pkg/sso/sso.go

```diff
@@ -0,0 +1,100 @@
+package sso
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"golang.org/x/oauth2"
+
+	"github.com/bacalhau-project/bacalhau/pkg/config/types"
+)
+
+// DeviceCodeResponse represents the response from the device authorization endpoint
+// This is our own representation of the device code response
+type DeviceCodeResponse struct {
+	DeviceCode              string `json:"device_code"`
+	UserCode                string `json:"user_code"`
+	VerificationURI         string `json:"verification_uri"`
+	VerificationURIComplete string `json:"verification_uri_complete"`
+	ExpiresIn               int    `json:"expires_in"`
+	Interval                int    `json:"interval"`
+	Message                 string `json:"message"`
+}
```

Comment:
not sure if it matters, but all our APIs are following title case

Reply:
@wdbaruni these are forced on us by the RFC https://datatracker.ietf.org/doc/html/rfc8628#section-3.2